### ISSUE TITLE
Asking some questions about training parameters using softmax script

### ISSUE THREAD

**Issue by neklom** [davidsandberg/facenet#1034]
Hi, I have some questions about the training parameters in train_softmax.py script and I hope someone could answer and help me.

For batch size we have default = 90 and number of batches per epoch is 1000 which is a I believe is 90*1000 = 90,000. While I'm using CASIA Webface dataset, it contains +450K images and after using --validation_set_split_ratio to split the data, I have like 425K images in training set now. As I know, in each epoch we should pass all the images for training, while having batch_size=90 and epoch_size=1000 we just pass 90K images in a single epoch. Is this normal or does it pass other images in other epochs ? Should I change those parameters to pass all images ?

And for the three parameters for data augmentation : --random_crop, --random_rotate, --random_flip, while using those parameters do we increase the batch size, like using random flip would give us horizontal flipping of training images so we double the size of images here, am i right ? Same for rotating and cropping.

A final question is about the center loss factor, if we use this loss function, how is it performed while using softmax loss too ?

Could somebody explain me more please and thanks !

**Comment by vkasojhaa** [comment]
@neklom The parameter epoch_size should rather be named evaluate_every_n_steps or similar since it has nothing to do with the number of examples in the dataset. You can refer to davidsandberg's answers #374, #158

**Comment by neklom** [comment]
@vkasojhaa Hi and thanks for your replay it helped me understand more.

The part of the center loss isn't really clear for me. Having center loss combined with the softmax loss, is this implementation similar to the center loss paper "A Discriminative Feature Learning Approach for Deep Face Recognition" ?

And thanks
